pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -658,9 +659,15 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	// we are passing false in case of osd and prepare pods because we don't want to overlap placement of osd on PVC's and non-PVC's.
 	p := cephv1.GetOSDPlacement(c.spec.Placement)
 	if osdProps.onPVC() {
+		// apply storageClassDeviceSet Placement
 		osdProps.placement.ApplyToPodSpec(&deployment.Spec.Template.Spec, false)
+		// apply placement.All() if it is not empty
+		if !reflect.DeepEqual(c.spec.Placement.All(), rookv1.Placement{}) {
+			p.ApplyToPodSpec(&deployment.Spec.Template.Spec, false)
```

Comment:
Wait, I'm not seeing how this works. If both `all` and `osd` sections are specified in the placement, they would both be applied here. But we only want to apply the `all`, and **not** the `osd` section.

Reply:
I still feel like this is the right approach. Did this not work? And you could delete line 660 for `GetOSDPlacement()`.
```go
if osdProps.onPVC() {
	// Merge the "all" and storageClassDeviceSet placement
	p := c.spec.Placement.All().Merge(*osdProps.preparePlacement)
	p.ApplyToPodSpec(&podSpec, false)
} else {
	// Merge the "all" and "osd" placement
	p := cephv1.GetOSDPlacement(c.spec.Placement)
	p.ApplyToPodSpec(&podSpec, false)
}
```